mindsdb/interfaces/datastore/datastore.py

```diff
@@ -26,7 +25,20 @@ def __init__(self):
     def get_analysis(self, name):
         datasource_record = session.query(Datasource).filter_by(company_id=self.company_id, name=name).first()
         if datasource_record.analysis is None:
-            datasource_record.analysis = json.dumps(self.mindsdb_native.analyse_dataset(self.get_datasource_obj(name)))
+            try:
+                original_process_title = setproctitle.getproctitle()
+                setproctitle.setproctitle('mindsdb_native_process')
+            except Exception:
+                pass
+
+            analysis = self.mindsdb_native.analyse_dataset(self.get_datasource_obj(name))
+
+            try:
+                setproctitle.setproctitle(original_process_title)
+            except Exception:
+                pass
```

Comment:
This approach can actually be used to "guard" predict calls as well (as part of the native interface)

Reply:
I added this to 'predict' method, but actually it not called in separate thread/process at this moment.